cli/src/semgrep/semgrep_core.py

```diff
@@ -35,18 +43,97 @@ def compute_executable_path(exec_name: str) -> Optional[str]:
 
 
 class SemgrepCore:
+    # Path to either the bridge module .so file or the semgrep-core
+    # executable, or None if we could not find either or have not looked
+    # yet.
     _SEMGREP_PATH_: Optional[str] = None
+
     _DEEP_PATH_: Optional[str] = None
 
+    # Reference to the bridge module if we are using it.
+    _bridge_module: Optional[types.ModuleType] = None
+
+    @classmethod
+    def executable_path(cls) -> str:
+        """
+        Return the path to the semgrep stand-alone executable binary,
+        *not* the Python module.  This is intended for unusual cases
+        that the module plumbing is not set up to handle.
+
+        Raise Exception if the executable is not found.
+        """
+        # This method does not bother to cache its result since it is
+        # expected to not be a part of any critical path.
+        ret = compute_executable_path("semgrep-core")
+        if ret is None:
+            raise Exception("Could not locate semgrep-core binary")
+        return ret
+
     @classmethod
     def path(cls) -> str:
+        """
+        Return the path to the semgrep binary, either the Python module
+        or the stand-alone program.  Raise Exception if neither is
+        found.
+        """
         if cls._SEMGREP_PATH_ is None:
-            cls._SEMGREP_PATH_ = compute_executable_path("semgrep-core")
+            # Try the module first (unless disabled).
+            if not os.getenv("SEMGREP_DISABLE_BRIDGE"):
```

Comment:
I'd rather make the fork/exec be the default code path, and switch to the bridge only when some environment variables are set (SEMGREP_USE_BRIDGE=1).

Reply:
Done in 4130f06e5a7807001855d20dcd2767ec028e96e8.